godot-core/src/builtin/plane.rs

```diff
@@ -202,4 +321,36 @@ mod test {
 
         crate::builtin::test_utils::roundtrip(&plane, expected_json);
     }
+
+    /// Tests plane methods.
+    #[test]
+    fn tests_methods() {
+        // Random planes including (0.0, 0.0, 0.0).
+        let a: Plane = Plane::new(Vector3::new(1.0, 2.0, 3.0).normalized(), 0.0);
+        let b: Plane = Plane::new(Vector3::new(0.5, 2.1, 7.1).normalized(), 0.0);
+        let c: Plane = Plane::new(Vector3::new(4.3, 9.2, 3.14).normalized(), 0.0);
+        let d: Vector3 = Vector3::new(0.0, 0.0, 0.0);
+        let e: Vector3 = Vector3::new(0.0, 0.0, 1.0);
+        assert_eq!(a.distance_to(d), 0.0);
+        assert_eq!(a.get_center(), d);
+        assert_eq!(a.has_point(d, None), true);
+        assert_eq!(match a.intersect_3(&b, &c) {
+            Some(x) => x,
+            None => Vector3::new(1.0, 1.0, 1.0),
+        }, d);
+        assert_eq!(a.is_finite(), true);
+        assert_eq!(a.is_point_over(d), false);
+        assert_eq!(a.is_point_over(e), true);
+        assert_eq!(a.is_point_over(-e), false);
+        assert_eq!(match a.intersects_ray(d, e) {
+            Some(x) => x,
+            None => Vector3::new(1.0, 1.0, 1.0),
+        }, d);
+        assert_eq!(match a.intersects_segment(d, e) {
+            Some(x) => x,
+            None => Vector3::new(1.0, 1.0, 1.0),
+        }, d);
+        assert_eq!(a.is_equal_approx(&a.normalized()), true);
+        assert_eq!(a.project(Plane::new(a.normalized().normal, 3.0).get_center()), a.get_center());
+    }
```

Comment:
It would probably make sense to group tests into related functionality; in some cases even one test per method. This helps identify test failures more easily and link them to a concrete part of the API.

For example:
* `test_normalized`
* `test_intersection`
* test_spatial_relations`
* ...

Reply:
I'm currently rewriting the tests, and I'm wondering whether comments like below to provide explanations are too excessive or not.

```rust
// Two coincident planes with an unrelated third plane would intersect in every point along the
// intersection line, thus no unique solutions.
assert_eq!(match plane_a.intersect_3(&plane_a, &plane_b) {
    Some(x) => x,
    None => fail,
}, fail);
// Three planes that intersects each other in a common line would intersect in every point along
// the line, thus no unique solutions.
assert_eq!(match plane_a.intersect_3(&plane_f, &plane_g) {
    Some(x) => x,
    None => fail,
}, fail);
```

Reply:
No, I think it's great!

I'm not sure about the code though 🙂 
Instead of 
```rs
assert_eq!(plane_a.intersect_3(&plane_f, &plane_g) {
    Some(x) => x,
    None => fail,
}, fail);
```
do you not just mean this?
```rs
assert_eq!(plane_a.intersect_3(&plane_f, &plane_g), None);
// or
assert!(plane_a.intersect_3(&plane_f, &plane_g).is_none());
```

Reply:
Oh yeah, that could work too 😂

Originally, that fail is a random vector to designate a fail, how I hadn't consider just comparing the `None` is beyond me :) Thanks for the better implementation!